backend-tests/tests/test_inventory.py

```diff
@@ -281,60 +320,285 @@ def logger(self):
+                assert len(test_case["response"]) == len(body), (
+                    "Unexpected number of results: %s != %s"
+                    % (
+                        [dev["id"] for dev in test_case["response"]],
+                        [dev["id"] for dev in body],
+                    )
+                )
 
-        useradmm = ApiClient(useradm.URL_MGMT)
-        devauthd = ApiClient(deviceauth.URL_DEVICES)
-        devauthm = ApiClient(deviceauth.URL_MGMT)
-        invd = ApiClient(inventory.URL_DEV)
-        invm_v2 = ApiClient(inventory_v2.URL_MGMT)
+                if len(test_case["response"]) > 0:
+                    if "sort" not in test_case["request"]:
+                        body = sorted(body, key=lambda dev: dev["id"])
+                        test_case["response"] = sorted(
+                            test_case["response"], key=lambda dev: dev["id"]
+                        )
 
-        # Initialize devices and API tokens
-        users = [tenant.users[0] for tenant in tenants_users]
-        inventories = [
-            {"version": "v1.0", "grp1": "foo", "idx": 0},
-            {"version": "v2.0", "grp1": "bar", "idx": 1},
-            {"version": "v3.0", "grp1": "baz", "idx": 2},
-            {"version": "v1.0", "grp2": "foo", "idx": 3},
-            {"version": "v2.0", "grp2": "bar", "idx": 4},
-            {"version": "v3.0", "grp2": "baz", "idx": 5},
-            {"version": "v1.0", "grp3": "foo", "idx": 6},
-            {"version": "v2.0", "grp3": "bar", "idx": 7},
-            {"version": "v3.0", "grp3": "baz", "idx": 8},
-        ]
+                    for i, dev in enumerate(test_case["response"]):
+                        assert (
+                            dev["id"] == body[i]["id"]
+                        ), "Unexpected device in response"
+                        assert_device_attributes(dev, body[i])
 
-        tenant = tenants_users[0]
-        tenant.devices = []
-        for inv in inventories:
-            user = tenant.users[0]
-            utoken = useradmm.call(
-                "POST", useradm.URL_LOGIN, auth=(user.name, user.pwd)
-            ).text
-            assert utoken != ""
-
-            tenant.api_token = utoken
-            device = make_accepted_device(
-                devauthd, devauthm, utoken, tenant_token=tenant.tenant_token
-            )
-            tenant.devices.append(device)
+    def test_search_v2_internal(self):
+        """
+        Tests the internal v2/{tenant_id}/filters/search endpoint.
+        This test along with the former covers all allowed operation
+        types.
+        """
 
-            attrs = self.dict_to_inventoryattrs(inv)
-            rsp = invd.with_auth(device.token).call(
-                "PATCH", inventory.URL_DEVICE_ATTRIBUTES, body=attrs
-            )
-            assert rsp.status_code == 200
-            device.inventory = inv
+        invm_v2 = ApiClient(inventory_v2.URL_INTERNAL)
 
         test_cases = [
```

Comment:
This has changed since the tests were written, now search is also available in open-source. I changed the tenant for the first test case to the open-source tenant just to include all plans.